schemas/input/csl-data.json

```diff
@@ -4,425 +4,477 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "original-author": {
+          "description": "[Deprecated - Use `author` on the `original` related item instead]",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "performer": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "producer": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "recipient": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "reviewed-author": {
+          "description": "[Deprecated - Use `author` on the `reviewed` related item instead]",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "translator": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/name-variable"
+          }
+        },
+        "accessed": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "available-date": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "container": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "event-date": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "issued": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "original-date": {
+          "description": "[Deprecated - Use a date property on the `original` related item instead]",
+          "$ref": "#/definitions/date-variable"
+        },
+        "submitted": {
+          "$ref": "#/definitions/date-variable"
+        },
+        "abstract": {
+          "type": "string"
+        },
+        "annote": {
+          "type": "string"
+        },
+        "archive": {
+          "type": "string"
+        },
+        "archive_collection": {
+          "type": "string"
+        },
+        "archive_location": {
+          "description": "[Deprecated - Q: what TODO?]",
+          "type": "string"
+        },
+        "archive-place": {
+          "description": "[Deprecated - Q: what TODO?]",
+          "type": "string"
+        },
+        "authority": {
+          "type": "string"
+        },
+        "call-number": {
+          "type": "string"
+        },
+        "chapter-number": {
+          "type": ["string", "number"]
+        },
+        "citation-number": {
+          "type": ["string", "number"]
+        },
+        "citation-label": {
+          "type": "string"
+        },
+        "collection-number": {
+          "type": ["string", "number"]
+        },
+        "collection-title": {
+          "type": "string"
+        },
+        "container-title": {
```

Comment:
With respect to `contianer-` variables, the principle reason to connect it with the `@related` structure is to enable a bibliography entry like this:

Jones, A. A. (2010). This is a book chapter. In Wilson (2010), pp. 100–150.
Smith, B. B. (2010). This is another chapter in the same book. In Wilson (2020), pp. 180–210.
Wilson, E. E. (Ed.). (2020). _This is the book holding the chapters._ Publisher. https://doi.org/10.1234/56789

Given this fairly rare need for a hierarchical structure, I'm inclined to do this:

```
        "container": {
          "title": "Container holding the item",
          "description": "Used to link an item to its container item.",
          "$ref": "#/definitions/related-id"
        },
```

That is, `container` only contains an `id` or `citekey`, not a full `bibitem`. The `container-title`, etc. for the item are entered as flat variables.

Reply:
The other situation that has been raised is for things like distinguishing if a `translator` is the translator of the item (e.g., a specific chapter) or of the whole book. I'm skeptical that this is a distinction we should try to make. Bibliographic metadata seem to rarely make this distinction (the translator of either the chapter or whole book is just listed as "translator"), so I don't think that styles can rely on a `translator` variable to mean the item translator and `container-translator` to mean the translator of the whole book. Same with editor. We would almost always want to assume that editor/translator refers to the role for the whole container.

A potentially more reliable translator heuristic might be: "If the there is no editor or if editor and translator are the same [i.e., `if variable="editor-translator"`], then `translator` is the translator of the whole container; if there is an editor who is different from the translator, then `translator` is the translator of the specific item." But even that seems too unreliable.

Reply:
> With respect to `contianer-` variables, the principle reason to connect it with the `@related` structure is to enable a bibliography entry like this:

If we had started with this structured, "container" and "collection" would likely be relations.

Consider the example of a chapter in a book, where you cite both. For that case, there's value in allowing the full relation.

It's worth discussing, but I did not include it, because it would make this more complicated.

Reply:
I appreciate that, but the flat data model allows a lot of valuable flexibility. For example, bibitem requires a type field. What is the type for `collection`—book series, lecture series, etc.? We don’t currently have types for these and probably don’t need them except as types for this internal hierarchical node. For containers, currently container-title can be used for all manner do containers (presentation sessions, books, journals)—constraining them to be specific enumerated types is less valuable than regarding the container as an container that need to be included on the cited item metadata—something that is specific to the item.